src/webgpu/api/operation/render_pipeline/culling_tests.spec.ts

```diff
@@ -150,36 +278,82 @@ TODO: check the contents of the depth and stencil outputs [2]
         },
         primitive: {
-          topology: t.params.primitiveTopology,
-          frontFace: t.params.frontFace,
-          cullMode: t.params.cullMode,
+          topology,
+          frontFace,
+          cullMode,
         },
-        depthStencil: depthTexture
-          ? { format: t.params.depthStencilFormat as GPUTextureFormat }
-          : undefined,
+        depthStencil,
       })
     );
-
-    pass.draw(6, 1, 0, 0);
+    pass.draw(6);
     pass.end();
 
     t.device.queue.submit([encoder.finish()]);
 
     // front facing color is green, non front facing is red, background is blue
-    const kCCWTriangleTopLeftColor = faceColor('ccw', t.params.frontFace, t.params.cullMode);
-    t.expectSinglePixelIn2DTexture(
-      texture,
-      format,
-      { x: 0, y: 0 },
-      { exp: kCCWTriangleTopLeftColor }
-    );
+    const kCCWTriangleTopLeftColor = faceColor('ccw', frontFace, cullMode);
+    const kCWTriangleBottomRightColor = faceColor('cw', frontFace, cullMode);
+    checkCornerPixels(t, texture, format, kCCWTriangleTopLeftColor, kCWTriangleBottomRightColor);
 
-    const kCWTriangleBottomRightColor = faceColor('cw', t.params.frontFace, t.params.cullMode);
-    t.expectSinglePixelIn2DTexture(
-      texture,
-      format,
-      { x: size - 1, y: size - 1 },
-      { exp: kCWTriangleBottomRightColor }
-    );
-    // [2]: check the contents of the depth and stencil outputs
+    if (depthTexture) {
+      // draw a triangle that covers all of clip space in yellow at the same depth
+      // as the previous triangles with the depth test set to 'less'. We should only
+      // draw yellow where the previous triangles did not.
+      depthStencilAttachment!.depthLoadOp = 'load';
+
+      if (haveStencil) {
+        depthStencilAttachment!.stencilLoadOp = 'load';
+        depthStencil!.stencilFront!.passOp = 'keep';
+        depthStencil!.stencilBack!.passOp = 'keep';
+      }
+
+      const k2ndDrawColor = new Uint8Array([255, 255, 0, 255]);
+
+      const isTopLeftCulled = faceIsCulled('ccw', frontFace, cullMode);
+      const kExpectedTopLeftColor = isTopLeftCulled ? k2ndDrawColor : kCCWTriangleTopLeftColor;
+
+      const isBottomRightCulled = faceIsCulled('cw', frontFace, cullMode);
+      const kExpectedBottomRightColor = isBottomRightCulled
+        ? k2ndDrawColor
+        : kCWTriangleBottomRightColor;
+
+      drawFullClipSpaceTriangleAndCheckCornerPixels(
+        t,
+        texture,
+        format,
+        topology,
+        k2ndDrawColor,
+        depthStencil as GPUDepthStencilState,
+        depthStencilAttachment as GPURenderPassDepthStencilAttachment,
+        kExpectedTopLeftColor,
+        kExpectedBottomRightColor
+      );
+
+      if (haveStencil) {
+        // draw a triangle that covers all of clip space in cyan with the stencil
+        // compare set to 'equal'. The reference value defaults to 0 so we should
+        // only render cyan where the first two triangles did not.
```

Comment:
note: Original TODO called for checking the depth write results as well, although I don't know that it would provide much useful coverage, so I'm ok with just dropping it if you agree.

Reply:
I'm fine with leaving it as is but happy to add checking actual values in another CL if you want it added